src/main/resources/lib/ssb/dataset/tbprocessor.ts

```diff
@@ -29,32 +30,73 @@ export function getTbprocessor(content: Content<DataSource>, branch: string): Da
   return null
 }
 
-export function fetchTbprocessorData(content: Content<DataSource>): TbmlData | null {
-  const baseUrl: string = app.config && app.config['ssb.tbprocessor.baseUrl'] ? app.config['ssb.tbprocessor.baseUrl'] : 'https://i.ssb.no/tbprocessor'
-  let data: TbmlData | null = null
-  if (content.data.dataSource) {
-    try {
-      const dataSource: DataSource['dataSource'] = content.data.dataSource
-      if (dataSource._selected && dataSource.tbprocessor && dataSource.tbprocessor.urlOrId) {
-        let url: string = `${baseUrl}/process/tbmldata/${getTbprocessorKey(content)}`
-        if (isUrl(dataSource.tbprocessor.urlOrId)) {
-          url = dataSource.tbprocessor.urlOrId
-        }
-        data = getTbmlData(url, content._id)
-      }
-    } catch (e) {
-      const message: string = `Failed to fetch data from tbprocessor: ${content._id} (${e})`
-      logUserDataQuery(content._id, {
+function hasTBProcessorDatasource(content: Content<DataSource>): string | undefined {
+  return content.data.dataSource &&
+    content.data.dataSource._selected &&
+    content.data.dataSource.tbprocessor &&
+    content.data.dataSource.tbprocessor.urlOrId
+}
+
+function tryRequestTbmlData(url: string, contentId?: string): TbmlData | null {
+  try {
+    return getTbmlData(url, contentId)
+  } catch (e) {
+    const message: string = `Failed to fetch data from tbprocessor: ${contentId} (${e})`
+    if (contentId) {
+      logUserDataQuery(contentId, {
         file: '/lib/ssb/dataset/tbprocessor.ts',
         function: 'fetchTbprocessorData',
         message: Events.REQUEST_COULD_NOT_CONNECT,
         info: message,
         status: e
       })
-      log.error(message)
     }
+    log.error(message)
+  }
+  return null
+}
+
+function tryRequestTbmlSourceList(url: string, contentId?: string): TbmlSourceList | null {
+  try {
+    return getTbmlSourceList(url)
+  } catch (e) {
+    const message: string = `Failed to fetch source list from tbprocessor: ${contentId} (${e})`
+    if (contentId) {
+      logUserDataQuery(contentId, {
+        file: '/lib/ssb/dataset/tbprocessor.ts',
+        function: 'tryRequestTbmlSourceList',
+        message: Events.REQUEST_COULD_NOT_CONNECT,
+        info: message,
+        status: e
+      })
+    }
+    log.error(message)
   }
-  return data
+  return null
+}
+
+function getDataAndMetaData(content: Content<DataSource>): TbmlData | null {
+  const baseUrl: string = app.config && app.config['ssb.tbprocessor.baseUrl'] ? app.config['ssb.tbprocessor.baseUrl'] : 'https://i.ssb.no/tbprocessor'
```

Comment:
Endre fallback to https://i.utv.ssb.no/tbprocessor or https://i.test.ssb.no/tbprocessor

Reply:
Default is set to test environment.